### PR TITLE
On the fly docs - add telemetry regarding documentation comments

### PR DESCRIPTION
**Pull request by akhera99**
Adds some telemetry to see if documentation comments impact the number of users clicking "tell me more" link.

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/QuickInfo/OnTheFlyDocsViewFactory.cs [hunk 1]
```diff
@@ -60,6 +60,8 @@ public OnTheFlyDocsViewFactory(IViewElementFactoryService factoryService, IAsync
             throw new InvalidOperationException("QuickInfoSession is null");
         }
 
+        OnTheFlyDocsLogger.LogShowedOnTheFlyDocsLink();
```
Comment (genlu) [inline src/EditorFeatures/Core.Wpf/QuickInfo/OnTheFlyDocsViewFactory.cs:63]: Could we log whether a symbol has doc-comment here? That would provide us the numbers regardless of whether user clicked the link or not, right?

File: src/Features/Core/Portable/QuickInfo/OnTheFlyDocsElement.cs [hunk 2]
```diff
@@ -12,9 +12,13 @@ namespace Microsoft.CodeAnalysis.QuickInfo;
 /// <param name="symbolSignature">formatted string representation of a symbol/></param>
 /// <param name="declarationCode">the symbol's declaration code</param>
 /// <param name="language">the language of the symbol</param>
-internal sealed class OnTheFlyDocsElement(string symbolSignature, ImmutableArray<string> declarationCode, string language)
+/// <param name="hasComments">whether the symbol has comments</param>
+internal sealed class OnTheFlyDocsElement(string symbolSignature, ImmutableArray<string> declarationCode, string language, bool hasComments)
```
Comment (genlu) [inline src/Features/Core/Portable/QuickInfo/OnTheFlyDocsElement.cs:16]: Maybe also differentiate source symbols and metadata symbols?

File: src/EditorFeatures/Core.Wpf/QuickInfo/OnTheFlyDocsView.xaml.cs [hunk 3]
```diff
@@ -206,8 +206,20 @@ public void RequestResults()
...
+            m["HasDocumentationComments"] = _onTheFlyDocsElement.HasComments;
         }, LogLevel.Information));
 
+        OnTheFlyDocsLogger.LogOnTheFlyDocsResultsRequested();
+        if (_onTheFlyDocsElement.HasComments)
+        {
+            Logger.Log(FunctionId.CodeAnalysisService_CalculateDiagnosticsAsync, KeyValueLogMessage.Create(m =>
+            {
+                m["SymbolHeaderText"] = _onTheFlyDocsElement.SymbolSignature;
+            }, LogLevel.Information));
```
Comment (genlu) [inline src/EditorFeatures/Core.Wpf/QuickInfo/OnTheFlyDocsView.xaml.cs:218]: what's this log message for?

Comment (akhera99) [reply src/EditorFeatures/Core.Wpf/QuickInfo/OnTheFlyDocsView.xaml.cs:218]: Oh I meant to change that

File: src/Features/CSharp/Portable/QuickInfo/CSharpSemanticQuickInfoProvider.cs [hunk 4]
```diff
@@ -181,6 +183,30 @@ protected override NullableFlowState GetNullabilityAnalysis(SemanticModel semant
             return sourceText.GetSubText(new Text.TextSpan(span.Start, Math.Min(maxLength, span.Length))).ToString();
         }).ToImmutableArray();
 
-        return new OnTheFlyDocsElement(symbol.ToDisplayString(), symbolStrings, symbol.Language);
+        return new OnTheFlyDocsElement(symbol.ToDisplayString(), symbolStrings, symbol.Language, hasDocumentationComment);
+
+        bool HasDocumentationComment(ImmutableArray<SyntaxReference> references)
```
Comment (genlu) [inline src/Features/CSharp/Portable/QuickInfo/CSharpSemanticQuickInfoProvider.cs:188]: when creating a tooltip in QuickInfoUtilities.CreateQuickInfoItemAsync, it looks like we know whether if contains  doc-comment, is it possible log there instead so we don't have to check the syntax?

File: src/Features/CSharp/Portable/QuickInfo/CSharpSemanticQuickInfoProvider.cs [hunk 5]
```diff
@@ -168,13 +168,20 @@ protected override NullableFlowState GetNullabilityAnalysis(SemanticModel semant
             return null;
         }
 
+        if (symbol.MetadataToken != 0)
```
Comment (genlu) [inline src/Features/CSharp/Portable/QuickInfo/CSharpSemanticQuickInfoProvider.cs:171]: so this does what we want? Great!

**Review by genlu — APPROVED** [review]

**Pull request merged**